src/sentry/api/endpoints/user_details.py

```diff
@@ -75,7 +75,11 @@ class UserOptionsSerializer(serializers.Serializer):
 
 class BaseUserSerializer(CamelSnakeModelSerializer):
     def validate_username(self, value):
-        if User.objects.filter(username__iexact=value).exclude(id=self.instance.id).exists():
+        if (
+            User.objects.filter(username__iexact=value)
+            .exclude(id=self.instance.id if hasattr(self.instance, "id") else 0)
```

Comment:
What's the reasoning behind defaulting to 0, is that the supposed default instance id or something?

Reply:
0 can't be a pk (1-indexed). Seemed simpler than writing a big branching logic block.